Content.Shared/_RMC14/BulletBox/BulletBoxSystem.cs

```diff
@@ -63,20 +90,36 @@ private void OnTransferDoAfter(Entity<BulletBoxComponent> ent, ref BulletBoxTran
         args.Handled = true;
 
         var user = args.User;
+        var transfer = 0;
         var used = new Entity<RefillableByBulletBoxComponent?, BallisticAmmoProviderComponent?>(usedId, null, null);
         if (!CanTransferPopup(ent, user, ref used) || used.Comp2 == null)
             return;
 
-        var transfer = used.Comp2.Capacity - used.Comp2.Count;
-        if (transfer <= 0)
-            return;
+        if (!ent.Comp.TransferToBox)
+        {
+            transfer = used.Comp2.Capacity - used.Comp2.Count;
+            if (transfer <= 0)
+                return;
 
-        transfer = Math.Min(transfer, ent.Comp.Amount);
-        _gun.SetBallisticUnspawned((used, used.Comp2), used.Comp2.UnspawnedCount + transfer);
-        ent.Comp.Amount -= transfer;
-        Dirty(ent);
+            transfer = Math.Min(transfer, ent.Comp.Amount);
+            _gun.SetBallisticUnspawned((used, used.Comp2), used.Comp2.UnspawnedCount + transfer);
+            ent.Comp.Amount -= transfer;
+        }
+        else
+        {
+            transfer = ent.Comp.Max - ent.Comp.Amount;
+            if (transfer <= 0)
+                return;
 
-        _popup.PopupClient(Loc.GetString("rmc-bullet-box-transfer-done", ("amount", transfer), ("used", used)), user);
+            transfer = Math.Min(transfer, used.Comp2.Count);
+            _gun.SetBallisticUnspawned((used, used.Comp2), used.Comp2.UnspawnedCount - transfer);
+            ent.Comp.Amount += transfer;
+        }
+        if (_net.IsServer)
+        {
+            _popup.PopupEntity(Loc.GetString("rmc-bullet-box-transfer-done", ("amount", transfer), ("used", ent)), ent, user);
+        }
```

Comment:
Same here, this needs to be popup client, not net is server popup entity, or it won't be predicted